mipengine/controller/node_landscape_aggregator.py

```diff
@@ -428,13 +401,66 @@ def _update_data_models_with_aggregated_datasets(
                 max=dataset_cde.max,
             )
             data_models[data_model].values["dataset"] = new_dataset_cde
+    return data_models
+
+
+def _remove_incompatible_data_models(
+    federation_info: FederationInfo,
+) -> FederationInfo:
+    """
+    Each node has its own data models definition.
+    We need to check for each data model if the definitions across all nodes is the same.
+    If the data model is not the same across all nodes containing it, we log the incompatibility.
+    The data models with similar definitions across all nodes are returned.
+    Parameters
+    ----------
+        federation_info: FederationInfo
+    Returns
+    ----------
+        FederationInfo
+            The federation info without the incompatible data models
+    """
+    validation_dictionary = {}
+
+    incompatible_data_models = []
+    for node_id, dmi in federation_info.values.items():
+        for data_model, datasets_and_cdes in dmi.values.items():
+            if data_model in incompatible_data_models:
+                continue
+
+            _, cdes = datasets_and_cdes
+            if data_model in validation_dictionary:
+                valid_node_id, valid_cdes = validation_dictionary[data_model]
+                if not valid_cdes == cdes:
```

Comment:
What if there is a difference in the datasets field. Have you overloaded the equals operator?

Reply:
Yeah __eq__ is overloaded to cover the case